webrender/src/resource_cache.rs

```diff
@@ -935,6 +936,24 @@ impl ResourceCache {
                 let (actual_width, actual_height) =
                     compute_tile_size(image_descriptor, tile_size, tile);
 
+                if let Some(ref mut dirty) = dirty_rect {
+                    *dirty = match dirty.intersection(&DeviceUintRect::new(
+                            DeviceUintPoint::new(
+                                tile.x as u32 * tile_size as u32,
+                                tile.y as u32 * tile_size as u32
+                            ),
+                            DeviceUintSize::new(actual_width, actual_height),
+                    )) {
+                        Some(mut r) => {
+                            // we can't translate by a negative size so do it manually
```

Comment:
we could probably have a `Vector2D` for that offset?

Reply:
DeviceUintVector2D would still need positive values so I don't think that would work.